src/aatoolbox/utils/io.py

```diff
@@ -82,35 +83,40 @@ def parse_yaml(filename: Union[str, Path]) -> dict:
     return config
 
 
-def check_file_existence(func: F) -> F:
+@wrapt.decorator
+def check_file_existence(wrapped: F, instance, args, kwargs) -> F:
     """
     Don't overwrite existing data.
 
     Avoid recreating data if it already exists and if clobber not
-    toggled by user. Can be used to wrap a function whose first parameter
-    is a filepath.
+    toggled by user. Used to wrap functions that accept filepath
+    as a keyword argument.
 
     Parameters
     ----------
-    func : function
-        The function to wrap. The first parameter of this function must
-        be the filepath (of type Path), and it can also have an optional
+    wrapped : function
+        The function to wrap. The function must have "filepath" as
+        a keyword parameter, and it can also have an optional
         "clobber" boolean keyword parameter.
+    instance :
+        Object the wrapped function is bound to. Not used within, but
+        ensures that instance methods do not pass `self` to args.
+    args :
+        List of positional arguments.
+    kwargs :
+        Dictionary of keyword arguments
 
     Returns
     -------
     If filepath exists, returns filepath. Otherwise, returns the result of
     the decorated function.
-
     """
-
-    def wrapper(filepath: Path, *args, **kwargs):
-        if filepath.exists() and not kwargs.get("clobber", False):
-            logger.info(
-                f"File {filepath} exists and clobber set to False, "
-                f"using existing files"
-            )
-            return filepath
-        return func(filepath, *args, **kwargs)
-
-    return cast(F, wrapper)
+    filepath = kwargs.get("filepath")
```

Comment:
only thing is that this is now a kwarg. So if say you have
```
@check_file_existence
def _download(filepath):
  ....
``` 
and you call `_download(filepath)` instead of `_download(filepath=filepath)`, it gives an error. Which is not bad, but maybe we could return an easier understandable error message cause now it is `AttributeError: 'NoneType' object has no attribute 'exists'`

Reply:
Yeah, thought about that. Previously was using `kwargs.pop()` which provides clear error message but of course, we actually needed `"filepath"` still in the dictionary when we pass it later on to `wrapped()` (as an aside, took me days to notice that was happening and only with Monica's help eventually realized hahaha). Since `_download()` is internal, wasn't sure if we should do error checking here or just assume we construct in the correct way so that `download()` would always call it with a keyword argument?

Reply:
Addressed!

Reply:
Actually, why not use a `try` here, catch the key error, and re-raise with the improved error message?

Reply:
Not sure exactly what you mean. `kwargs.get()` doesn't raise an error message, it just returns `None`, hence the need to check. Don't want to use `.pop()`, which does raise an error message, because we want to keep the `kwarg` in the dict for passing to `wrapped`.

Reply:
You can just do:
```python
try:
    filepath = kwargs["filepath"]
except KeyError as err:
   raise KeyError("message") from err
```